jenkinsfile_check.sh

```
@@ -59,3 +59,27 @@
+        java_version=$(xsltproc remove-namespaces.xsl "$pom_file" > pom-no-namespaces.xml && mv pom-no-namespaces.xml "$pom_file" && xmllint --xpath "string(//plugin[artifactId='maven-compiler-plugin']/configuration/source)" "$pom_file")
+    fi
+    if [ -z "$java_version" ]; then
+        java_version=$(xsltproc remove-namespaces.xsl "$pom_file" > pom-no-namespaces.xml && mv pom-no-namespaces.xml "$pom_file" && xmllint --xpath "string(//plugin[artifactId='maven-compiler-plugin']/configuration/target)" "$pom_file")
+    fi
+    echo "$java_version"
+}
+
+# Function to extract Jenkins core version from pom.xml
+get_jenkins_core_version_from_pom() {
+    local pom_file=$1
+    local jenkins_core_version=$(xsltproc remove-namespaces.xsl "$pom_file" > pom-no-namespaces.xml && mv pom-no-namespaces.xml "$pom_file" && xmllint --xpath "string($pom_xml_jenkins_core_version_xpath)" "$pom_file")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)